### PR TITLE
[NITF] NITF Publisher with customizations for NTB

### PR DESCRIPTION
**Pull request by jerome-poisson**
This commit implement a new NTBNITF subscribe type and publish items
in NITF format with customizations for NTB.

SDNTB-229

### CODE REVIEW

**Comment by jerome-poisson** [comment]
https://github.com/superdesk/superdesk-core/pull/462 need to be merged before this one can be merged

**Comment by mugurrus** [comment]
:+1:

File: server/ntb/publish/ntb_nitf.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        :param str format_type:
+        :param dict article:
+        :return: True if article can formatted else False
+        """
+        return format_type == 'ntbnitf' and article[ITEM_TYPE] == CONTENT_TYPE.TEXT
+
+    def _format_tobject(self, article, head):
+        category = ''
+        for s in article.get('subject', []):
+            if s.get('scheme') == 'category':
```
Comment (ioanpocol) [inline server/ntb/publish/ntb_nitf.py]: Category is a custom field added for NTB. Can we avoid this check and make it more generic?
For example we can test is scheme is not subject to get qcode.

File: server/ntb/publish/ntb_nitf.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                category = s['qcode']
+        return SubElement(head, 'tobject', {'tobject.type': category})
+
+    def _format_docdata(self, article, docdata):
+        super()._format_docdata(article, docdata)
+        if 'slugline' in article:
+            SubElement(docdata, 'du-key', attrib={'version': '1', 'key': article['slugline']})
+
+    def _format_subjects(self, article, tobject):
+        subjects = [s for s in article.get('subject', []) if s.get("scheme") == "subject_custom"]
```
Comment (ioanpocol) [inline server/ntb/publish/ntb_nitf.py]: Subject_custom is a scheme added for NTB only. We have to avoid to use custom fields in code.

**Comment by ioanpocol** [comment]
:+1: 

If the references to custom fields are avoided, this class can be added to core.

**Comment by jerome-poisson** [comment]
@mugurrus indeed, I'll add some before merging
@ioanpocol  normally this class is only handling NTB specific stuff, the generic one are in the core NITFFormatter class. But that true that the regex and method in _format_body_content could be useful for future subclasses.

**Pull request merged**